File: Contests/2021-22/CCSC_MW_21/Factor_Follies/factors.py
n, m = [int(i) for i in input().split()]
factors = [int(i) for i in input().split()]
multiples = [int(i) for i in input().split()]
ans = []

not_ans = False
for mult in multiples:
    for fact in factors:
        if mult % fact != 0:
            not_ans = True
            break
    
    if not_ans:
        not_ans = False
        continue

    ans.append(mult)

print(" ".join([str(i) for i in ans]))

File: Contests/2021-22/CCSC_MW_21/Factor_Follies/Factors.java
import java.util.Scanner;
public class Factors {

    public static void main(String [] args) {
        Scanner sc = new Scanner(System.in);
        //read input
        int n = sc.nextInt();
        int m = sc.nextInt();;
        int [] factors = new int [n];
        int [] multiples = new int [m];
        for(int i = 0; i<n; i++){
            factors[i] = sc.nextInt();
        } 
        for(int i = 0; i<m; i++){
            multiples[i] = sc.nextInt();
        } 
        //check each factor for each multiple
        for(int i = 0; i<multiples.length; i++){
            boolean valid = true;
            for(int j = 0; j<factors.length; j++){
                if (multiples[i]%factors[j] != 0) {
                    valid = false;  
                    break;
                }
            }
            if(valid) System.out.print(multiples[i]+" ");
        }   
    }  
}
